### PR TITLE
Implemented Waveshare 7.5in B V3

### PR DESCRIPTION
**Pull request by lucasprim**
- Implemented output instructions that updates the Waveshare 7.5" B V3
- Updates now take ~10 less seconds than using 7.5" B V2
- B/W only

# What does this implement/fix?

This implements proper code to make the Waveshare 7.5" model B, V3. [Specs](https://www.waveshare.com/w/upload/8/8c/7.5inch-e-paper-b-v3-specification.pdf)
Even though the Model B V2 code works for the V3, it takes about 15 seconds to update the screen, and in a flickery way.
This code, based on the Waveshare [E-Paper ESP32 Demo](https://www.waveshare.com/wiki/E-Paper_ESP32_Driver_Board#WiFi_Demo) is capable of updating the screen in a couple of seconds in a reliable way.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other

**Related issue or feature (if applicable):** Not Applicable

**Pull request in [esphome-docs](https://github.com/esphome/esphome-docs) with documentation (if applicable):** esphome/esphome-docs#3111

## Test Environment

- [X] ESP32
- [ ] ESP32 IDF
- [ ] ESP8266
- [ ] RP2040

## Example entry for `config.yaml`:

```yaml
# Example config.yaml
font:
  - file:
      type: gfonts
      family: Inter
      weight: 900
    id: font_main_title
    size: 24

spi:
  clk_pin: 13
  mosi_pin: 14
  id: epaper_display

display:
  - platform: waveshare_epaper
    id: epaper
    cs_pin: 15
    dc_pin: 27
    busy_pin: 25
    reset_pin: 26
    model: 7.50in-bV3
    update_interval: 30s
    reset_duration: 2ms
    rotation: 90°
    lambda: |-
      it.filled_circle(240, 200, 50);
      it.print(240, 400, id(font_main_title), TextAlign::CENTER, "HELLO WORLD!");
      it.filled_circle(240, 600, 50);
```

## Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Tests have been added to verify that the new code works (under `tests/` folder).

If user exposed functionality or configuration variables are added/changed:
  - [X] Documentation added/updated in [esphome-docs](https://github.com/esphome/esphome-docs).

### CODE REVIEW

**Review by FeSens — COMMENTED** [review]
👍

File: esphome/components/waveshare_epaper/waveshare_epaper.cpp [hunk 1]
```diff
@@ -1328,6 +1328,157 @@ void WaveshareEPaper7P5InBV2::dump_config() {
...
+  const uint32_t start = millis();
+  while (this->busy_pin_->digital_read()) {
+    this->command(0x71);
+    if (millis() - start > this->idle_timeout_()) {
+      ESP_LOGI(TAG, "Timeout while displaying image!");
+      return false;
+    }
+    delay(10);
+  }
+  delay(200);  // NOLINT
```
Comment (jesserockz) [inline esphome/components/waveshare_epaper/waveshare_epaper.cpp:1346]: Why is a further delay required after the pin has been confirmed?

**Review by jesserockz — APPROVED** [review]
One question below

File: esphome/components/waveshare_epaper/waveshare_epaper.cpp (hunk 1 above)
Comment (lucasprim) [reply esphome/components/waveshare_epaper/waveshare_epaper.cpp:1346]: That's a good question.

TBH, I'm simply mimicking the behavior i've found on the manufacturer's implementation, as follows:
image

I've not tested it thoroughly with different delays, but removing it has impacted the reliability of the display, in some cases displaying noisy results.

**Pull request merged**